src/backend/env/memory.rs

```diff
@@ -214,7 +237,8 @@ impl Allocator {
         Ok(start)
     }
 
-    fn free(&mut self, offset: u64, size: u64) -> Result<(), String> {
+    fn free(&mut self, offset: u64, len: u64) -> Result<(), String> {
+        let size = self.get_allocation_length(len);
```

Comment:
If block size changes between `alloc` and `free`, then this will lead to data corruption, so we need to be careful when changing the block size.

Reply:
Exactly, this is why we need to unload everything in post_upgrade first.

I plan to comment out the block changing logic after this upgrade as well.